src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -551,13 +551,35 @@ void MPS::apply_swap_internal(uint_t index_A, uint_t index_B, bool swap_gate) {
     // and the qubit at index_B (or index_A+1) is moved one position 
     //to the left
     std::swap(qubit_ordering_.order_[index_A], qubit_ordering_.order_[index_B]);    
+    // For logging purposes:
+    if (MPS::get_mps_log_data()) {
+      if (swap_gate)
```

Comment:
We're already in an `if (!swap_gate)`. So one of the two options:
1. Get out of the scope of the `if (!swap_gate)`. In this case, why do you distinguish between `swap_gate` true or false, and why would it affect the order in the call to `print_to_log_internal_swap` ?
2. Remove the `if (swap_gate)` here.

Reply:
You are right, the internal `if (swap_gate)` is redundant. I removed it.